src/Mutator/Cast/CastArray.php

```diff
@@ -65,24 +63,6 @@ public function canMutate(Node $node): bool
             return false;
         }
 
-        $parent = ParentConnector::getParent($node);
-
-        if ($parent instanceof Node\Stmt\Return_) {
-            $functionScope = $this->findFunctionScope($parent);
-
-            if ($functionScope !== null) {
-                if ($functionScope->getAttribute(ReflectionVisitor::STRICT_TYPES_KEY) === false) {
-                    return true;
-                }
-
-                $returnType = $functionScope->getReturnType();
-
-                if ($returnType instanceof Node\Identifier && $returnType->name === 'array') {
-                    return false;
-                }
-            }
-        }
-
-        return true;
```

Comment:
in all cast classes I factored out this common logic into the base-class to prevent duplication and ease maintenance